site/_posts/2022-10-30-spring-and-fusionAuth.md

```diff
@@ -0,0 +1,353 @@
+
+### Creating an application
+
+Click "Setup" under "Missing Application" and call your new app "Spring Example" or another name of your choice. It'll get a Client Id and Client Secret automatically—save these, as we'll use them in the code. Later, we'll set up a Java and Spring application that will run on `http://localhost:8080`, so configure the Authorized URLs accordingly. You should add:
+
+- `http://localhost:8080/login/oauth2/code/fusionauth` to the Authorized redirect URLs.
+- `http://localhost:8080/` to the Authorized request origin URL.
+- `http://localhost:8080/logout` to the Logout URL.
+  
+{% include _image.liquid src="/assets/img/blogs/spring-fusionauth/fusionauth-urlconf.png" alt="Configuring the application URLs in FusionAuth." class="img-fluid" figure=false %}
+
+To enable self-service registrations, select the "Registration" tab. Scroll down to "Self-service registration", and switch "Enabled" on. Choose "Basic" as the registration type. You can choose the login type as either "Email" or "Username" - we've used "Email" for this tutorial. You can also select any other fields you'd like to capture on registration - we've left them off. Your setup should look similar to this:
+
+{% include _image.liquid src="/assets/img/blogs/spring-fusionauth/fusionauth-registration-settings.png" alt="Configuring the registration options in FusionAuth" class="img-fluid" figure=false %}
+
+Click the Save button at the top right for your changes to take effect.
+
+## Setting up OpenID Connect (OIDC)
+
+Once the user has logged in via the FusionAuth application, we can retrieve their FusionAuth profile using the [OIDC](https://fusionauth.io/docs/v1/tech/oauth/) functionality provided by FusionAuth.
+
+To configure this to work properly with Spring, we need to set a couple JWT parameters in FusionAuth.
+
+First, head to your Tenant configuration in FusionAuth. If you haven't made and linked a custom Tenant for your application, you can edit the default Tenant. Under the "General" tab in your Tenant settings, set the "Issuer" field to a URL representing your domain. Note that it must be a fully qualified URL for Spring to process it correctly (i.e. include the protocol). This issuer field is used in the [JWT](https://jwt.io) created by FusionAuth when logging in.
+
+{% include _image.liquid src="/assets/img/blogs/spring-fusionauth/fusionauth-issuer-url.png" alt="Configuring the JWT issuer URL in FusionAuth." class="img-fluid" figure=false %}
+
+Then head over to your FusionAuth application and edit it. Choose the "JWT" tab on the settings page, and enable JWT by setting the toggle to on. Then configure the JSON web tokens with the following values:
+
+- "JWT duration": 3600
+- "Access Token signing key": Access token signing key generated for application <name> (RS256)
```

Comment:
Where did you create this key? I missed it.

Reply:
There is no key that needs to be created - The text refers to the option in the FusionAuth JWT settings
![Screenshot 2022-11-29 at 3 28 25 PM](https://user-images.githubusercontent.com/4689280/204541325-ca5183fb-ccdf-449c-a0e2-0c8c505e5790.png)